src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -117,7 +117,7 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
             var  evaluations = AnalyzeAllIncludedTemplates(template, parameters, templateFilePath, templateContext, "");
 
             // For each rule we don't want to report the same line more than once
```

Comment:
This code is a temporary solution until issue #228 is resolved, it shouldn't be needed if the issue is closed by this PR. It also shouldn't be needed to dedupe results in ConsoleReportWriterTests, SarifReportWriterTests and TemplateAnalyzerTests

Reply:
OK that makes sense I will try removing. I still think that the deduping of the results in the tests is needed because a result with multiple paths (i.e. bicep modules) will have multiple evaluations with identical results (except they technically differ by target file/reference path).

This is easy to verify for sure though by removing it and seeing if the tests still pass.

Reply:
I'm not sure if I follow, why identical results? Like the same line number but a different file path?

Reply:
Yes, if a bicep file has multiple references to another template with results, there would be multiple "paths" through the bicep to those issues--so in the results we would have an expression with identical results for each unique path to that template.

Reply:
I don't understand why we don't dedup all of those results then, why do we want to return multiple identical results?

Reply:
So since the last comment, I've removed the "stack" from the source location so effectively a result will look identical for a rule whether it's found as part of a module or scanned in the entrypoint file. Also I tried removing this code assuming that the late deduping would cover it and interestingly TA-000021 fails so I need to investigate that further.

Reply:
The result deduping code now is in the Analyzer.Reports. As reporting is done outside/after TemplateAnalyze.AnalyzeTemplate() that means either we refactor to dedupe in TemplateAnalyzer or just update the tests to expect the dupes since they will be deduped in the reporting code (where we have tests specifically for deduping still).

Reply:
I vote for the option where we don't need to expect duplicated results in the tests

Reply:
Especially because this also affects the built-in rule tests, it adds so much noise to repeat the description of each failed line number for each repeated result:
https://github.com/Azure/template-analyzer/blob/87e6e2bb8e9228521c855c857df2cdb257545a68/src/Analyzer.Core.BuiltInRuleTests/Tests/TA-000021/TA-000021.json

I believe it was one of the reasons for this issue: https://github.com/Azure/template-analyzer/issues/228

If we are closing it without fixing this I think we should at least open a new one

Reply:
That's interesting because I read this quote from #228: "_Reports_ could then do something simple like the following..." and concluded that the reports were the place to do the deduping. If we do want to do deduping in TemplateAnalyzer class, we will need deduping in the reports as we still need to dedupe across files with modules. So, I guess I am not sure what the original intent of where the deduping lives, but I am open to your notion that there's not dupes from TemplateAnalyzer itself. I think we could make use of distinct() here to clean up the temp deduping code I removed as one option.

Reply:
Okay, now I'm not so sure what was the original idea of the issue, but when I linked PR #262 to it I was referring to the duplications in the built-in rule tests

I'm not sure if I understood the last part either, do you mean adding a .Distinct() in the TestRunner to avoid repeated descriptions in the test files? That sounds good to me

Reply:
OK I see what you mean, I don't think that's what meant with your suggestion, but I like it better anyway to clean up the duplication in the built in rule tests. Will do this shortly.